platform/kube/cache.go

```diff
@@ -66,6 +75,9 @@ func (pc *PodCache) getPodByIP(addr string) (*v1.Pod, bool) {
 
 // labelsByIP returns pod labels or nil if pod not found or an error occurred
 func (pc *PodCache) labelsByIP(addr string) (model.Labels, bool) {
+	pc.rwMu.RLock()
+	defer pc.rwMu.RUnlock()
+
 	pod, exists := pc.getPodByIP(addr)
```

Comment:
double read lock?

Reply:
fixed PTAL

Reply:
+1, please revert lines 78-80, not needed.